### PR TITLE
Feature/default sort relevance

### PR DESCRIPTION
**Pull request by chrpr**
@barnabyalter @NYULibraries/finding-aids 

Hi Barnaby, et al.,

Here's a first cut at PR for story ID 71060102. 

A few questions: 
* Should I be doing any testing beyond confirming that the correct sort option is selected?
* Was it appropriate to follow the pattern used by "facets", keeping "dropdown" tests in a separate folder? 
* If yes to above, is there still a way to run just the tests in a specific file? When I try to select a file or a start line within that file, it says steps are undefined...
* Should I put the other related sort stories on separate PRs, or consolidate them here?

Thanks,
-Corey

### CODE REVIEW

**Comment by barnabyalter** [comment]
@chrpr Actually I think that this is one feature called `sorting.feature` and it includes one scenario for each sorting option and one for the default option. For your case I think there are two scenarios: one for the default and one for `When I select "Sort by relevance"`. The latter will hold up the functionality if, say, the default search option changes at some point.

File: features/dropdowns/relevance_default.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+  Feature: Default sort is relevance
+    In order to identify the most relevant results
+    As a researcher
+    I wan the default sort option to be Relevance
+
+  Scenario: Default sort option is Relevance
+    Given I am on the default search page
+    When I search on the phrase "bloch"
+    Then "Sort by relevance" should be selected for "sort" dropdown
```
Comment (barnabyalter) [inline features/dropdowns/relevance_default.feature]: @chrpr Sorting is only on the brief results screen so you could simplify assuming you're always on the brief results screen and then each story could be 

```
Given I am on the brief results screen
When I select "relevance" from the "Sort by" dropdown 
Then the results should be sorted by relevance
```

Where the last statement is just a feature-pointed way of checking that the dropdown is correctly selected.

Thoughts?

Comment (chrpr) [inline features/dropdowns/relevance_default.feature]: @barnabyalter Thanks for your feedback yesterday. This definitely makes sense to me. I'll move this up a level and call it sorting.feature, and start from "Given I am on the brief results screen".

I'm assuming I should actually have 2 here, though:
Given I am on the brief results screen
Then the results should be sorted by relevance

and 

Given I am on the brief results screen
When I select "relevance" from the "Sort by" dropdown 
Then the results should be sorted by relevance

This covers the default versus selected. And you think the "results should be sorted by relevance" should not actually look at the sorting, but rather just at whether the correct option is selected? 

An alternate approach would be to use a more broad search, like "archives", that returns more than one hit and confirm which is first for different options.

In any case, I'll post a few more commits to the PR later this afternoon, then move on to other stories tomorrow.

**Comment by chrpr** [comment]
@barnabyalter Pushed a few new commits which I think address your concerns. My question remains about whether we should actually verify that the sort option changes sort order. I can do this by switching to a broader search (archives), for which sort by relevance puts the record with "archives" in the title first, and sort by date moves it to second. Maybe that's another user story, though.

Let me know if you'd like to see anything else done differently here.

Also, @NYULibraries/finding-aids -- In case anyone else wants the answer to how to run just one cucumber test, you can say something like:
bundle exec cucumber features/facets/library_facet.feature -r features/

**Comment by barnabyalter** [comment]
@chrpr I think testing the actual sort relevancy is too difficult a story now considering the state of indexing and searching. Maybe when all is said and done we can add a few cakes that make sure things like exact title come up first, etc.

For now this looks good... merging

**Pull request merged**